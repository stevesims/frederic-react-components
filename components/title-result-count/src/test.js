import React from 'react';
import { shallow } from 'enzyme';
import { createMatchers } from 'jest-emotion';
import * as emotion from 'emotion';

import Component from '.';

expect.extend(createMatchers(emotion));

describe('TitleResultCount', () => {
  let wrapper;

  it('renders with no properties', () => {
    wrapper = shallow(<Component>Sample title</Component>);
  });

  it('renders style rules', () => {
    expect(wrapper).toHaveStyleRule('display', 'inline-block');
  });

  it('renders title', () => {
    expect(
      wrapper
        .find('Title')
        .childAt(0)
        .text(),
    ).toBe('Sample title');
  });

  it('renders with a count', () => {
    wrapper.setProps({ count: 3 });
    expect(
      wrapper
        .find('Styled(span)')
        .childAt(0)
        .text(),
    ).toBe('3');
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
