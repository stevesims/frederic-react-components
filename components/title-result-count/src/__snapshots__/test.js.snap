// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TitleResultCount matches snapshot 1`] = `
<Wrapper>
  <Title>
    Sample title
  </Title>
  <Styled(span)>
    3
  </Styled(span)>
</Wrapper>
`;
